src/target/stm32f4.c

```diff
@@ -615,7 +616,7 @@ static bool stm32f4_option_write(target *t, uint32_t *val, int count)
 	/* WRITE option bytes instruction */
 	if (((t->part_id == ID_STM32F42X) || (t->part_id == ID_STM32F46X) ||
 		 (t->part_id == ID_STM32F72X) || (t->part_id == ID_STM32F74X) ||
-		 (t->part_id == ID_STM32F76X)) && (count > 1))
+		 (t->part_id == ID_STM32F76X) || (t->part_id == ID_GD32F470)) && (count > 1))
```

Comment:
Just a heads-up that, when fix/clang-format-cleanup lands, this change will break and will need to go in the newly introduced helper function `stm32f4_opt_bytes_for` - this seems fine otherwise.

It's not your fault, but a note for future contributions, the parentheses around the comparison are considered harmful as they suppress a useful GCC warning about accidental assignment instead of comparison. We've stripped them on the clean up branch but we can see you're following the existing style here. We've been slowly transitioning the code to not including the extra parentheses for this reason and to improve clarity.

Reply:
That all sounds reasonable to me -- would it make sense for me to rebase this on that branch?

Reply:
Given we've got to pull that branch apart into 4-5 for PR'ing and the code changes have yet to be reviewed, we'd hold off. All the changes in question will land in 1.9 or early 1.10 at latest, so we'll try to remember to ping you on this PR once the breaking changes for this have landed.

Reply:
The breaking changes for this have landed so if you could please rebase on main, we will look at getting this merged ASAP.